dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,336 @@
+        try (InputStream is = IOUtils.toInputStream(bitstream2Content, CharEncoding.UTF_8)) {
+            bitstream2 = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream 2")
+                    .withDescription("description")
+                    .withMimeType(bitstream2MimeType)
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        String url = configurationService.getProperty("dspace.ui.url");
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json")
+                        .header("Accept", "application/linkset+json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))))
+                .andExpect(jsonPath("$.linkset[0].item[0].href",
+                        Matchers.hasToString(url + "/bitstreams/" + bitstream1.getID() + "/download")))
+                .andExpect(jsonPath("$.linkset[0].item[0].type",
+                        Matchers.hasToString(bitstream1MimeType)))
+                .andExpect(jsonPath("$.linkset[0].item[1].href",
+                        Matchers.hasToString(url + "/bitstreams/" + bitstream2.getID() + "/download")))
+                .andExpect(jsonPath("$.linkset[0].item[1].type",
+                        Matchers.hasToString(bitstream2MimeType)))
+                .andExpect(jsonPath("$.linkset[0].anchor",
+                        Matchers.hasToString(url + "/handle/" + item.getHandle())))
+                .andExpect(jsonPath("$.linkset[0].landingPage[0].href",
+                        Matchers.hasToString(url + "/entities/publication/" + item.getID())))
+                .andExpect(jsonPath("$.linkset[0].landingPage[0].type",
+                        Matchers.hasToString("text/html")));
+    }
+
+    @Test
+    public void findOneBitstreamJsonLinksets() throws Exception {
+        String bitstreamContent = "ThisIsSomeDummyText";
+        String bitstreamMimeType = "text/plain";
+
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Item Test")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .build();
+
+        Bitstream bitstream = null;
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+            bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream")
+                    .withDescription("description")
+                    .withMimeType(bitstreamMimeType)
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + bitstream.getID() + "/json")
```

Comment:
What about similar tests for Communities & Collections?  It appears those will also throw a 404.  But we should have a test which verifies that's the expected behavior.

Reply:
Additional tests has been added.